internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -567,12 +567,8 @@ func (a *Adapter) EnsureOverrideSnapshotValid() (controller.OperationResult, err
 }
 
 // EnsureGroupSnapshotExist is an operation that ensure the group snapshot is created for component snapshots
-// once a new component snapshot is created for an pull request and there are multiple existing PRs belonging to the same PR group
+// once a new component snapshot is created for an pull or push request and there are multiple existing PRs belonging to the same PR group
 func (a *Adapter) EnsureGroupSnapshotExist() (controller.OperationResult, error) {
-	if gitops.IsSnapshotCreatedByPACPushEvent(a.snapshot) {
```

Comment:
It is not possible to create reliable group Snapshots for push events, so I don't think we should remove this condition.

Reply:
+1, push event won't expect group snapshot.